Assets/UI/unitflagmanager_CQUI.lua

```diff
@@ -344,6 +350,73 @@ function OnUnitPromotionChanged( playerID : number, unitID : number )
     end
 end
 
+-- ===========================================================================
+--  CQUI modified OnLensLayerOn function
+--  If Religion Lens is enabled, call Refresh so non-religious units can be dimmed
+-- ===========================================================================
+function OnLensLayerOn( layerNum:number )
+    BASE_CQUI_OnLensLayerOn(layerNum);
+
+    if (layerNum == m_HexColoringReligion) then
+        m_IsReligionLensOn = true;
+        -- Call Refresh, which will cycle through all of the units visible to the current player
+        -- OnUnitVisibilityChanged will be called for every unit the player can see, where the
+        -- icon can be dimmed due to the religion lens being selected.
+        Refresh();
+    end
+end
+
+-- ===========================================================================
+--  CQUI modified OnLensLayerOff function
+--  If Religion Lens is disabled, call Refresh so non-religious units can be restored
+-- ===========================================================================
+function OnLensLayerOff( layerNum:number )
+    BASE_CQUI_OnLensLayerOff(layerNum);
+
+    if (layerNum == m_HexColoringReligion) then
+        m_IsReligionLensOn = false;
+        Refresh();
+    end
+end
+
+-- ===========================================================================
+--  CQUI modified OnUnitVisibilityChanged function
+--  Override basegame OnUnitVisibilityChanged to dim non-religious icons when the religion lens is displayed
```

Comment:
Why is this an override? Seems possible to make an extension - you get flagInstance from a call based on playerID and unitID.

Reply:
My thinking was to avoid doing the "work" of retrieving the flagInstance more than once, and because the unmodified function is short enough that there is low risk of breaking due to future Firaxis changes.

Reply:
This file is an extension, so all functions should (if possible) be extensions. Exceptions are allowed ofc, e.g. making an extension is super hard or impossible. But Imho how long or short a function is should not a criterion here.